var/spack/repos/builtin/packages/mono/package.py

```diff
@@ -74,6 +75,8 @@ def patch(self):
 
     def configure_args(self):
         args = []
-        li = self.spec["iconv"].prefix
-        args.append("--with-libiconv-prefix={p}".format(p=li))
+        if self.spec["iconv"].name == "libc":
+            args.append("--without-libiconv-prefix")
+        elif not is_system_path(self.spec["iconv"].prefix):
+            args.append("--with-libiconv-prefix={p}".format(p=self.spec["iconv"].prefix))
```

Comment:
I have looked at the configure and `mono` does not seem to use configure arguments `--without-libiconv-prefix` or `--with-libiconv-prefix`. When it searches for `libiconv` it uses default flags. If we want to change the path for `libiconv`,  we have to use `CPPFLAGS` or `LDFLAGS` environment variables.